### PR TITLE
Allow cancelling authentication and reloading window

### PR DESCRIPTION
**Pull request by peterbom**
As a temporary workaround for [this issue](https://github.com/microsoft/vscode/issues/211406), we can display a cancellable progress notification which reloads the window if the user clicks cancel.

This is really not ideal, and we should roll this back when the underlying issue is fixed, but it will serve to unblock any users who are left wondering why their sign-in attempt is not doing anything.

### CODE REVIEW

**Review by Tatsinnit — APPROVED** [review]
Looks good! Noticed few things and added few suggestions, thanks heaps. 🙏☕️

File: src/auth/azureSessionProvider.ts [hunk 1]
```diff
@@ -243,7 +247,28 @@ class AzureSessionProviderImpl extends VsCodeDisposable implements AzureSessionP
             }
 
             if (!session) {
-                session = await authentication.getSession(getConfiguredAuthProviderId(), scopes, options);
+                const authenticationPromise = authentication.getSession(getConfiguredAuthProviderId(), scopes, options);
```
[the remark this answers is not in the document]
Comment (peterbom) [reply src/auth/azureSessionProvider.ts:250]: Oops, yep, the plan was to use the promise in the `if` block. I'll change that.

File: src/auth/azureSessionProvider.ts [hunk 2]
```diff
@@ -243,7 +247,28 @@ class AzureSessionProviderImpl extends VsCodeDisposable implements AzureSessionP
...
+                const allowCancellation = options.silent !== true;
+                if (allowCancellation) {
```
[the remark this answers is not in the document]
Comment (peterbom) [reply src/auth/azureSessionProvider.ts:252]: No strong feelings either way - happy to change

**Comment by peterbom** [comment]
Having looked at the [fix going in to vscode](https://github.com/microsoft/vscode/pull/211495), I don't think we should merge this.

Once the next VS Code update is released (which I think could be anytime now), it will display its own cancellable notifications for this. If we release this change, we'd end up showing _two_ notifications. I think that's worse than the original bug (which [has been present](https://github.com/microsoft/vscode-azureresourcegroups/issues/736) in the Azure Resources extension for several months, apparently without serious impact).

**Comment by peterbom** [comment]
Closing as the underlying VS Code work item has now been completed.

**Pull request closed without merging**